ambry-router/src/main/java/com/github/ambry/router/OperationController.java

```diff
@@ -39,6 +39,7 @@
 import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.CopyOnWriteArrayList;
```

Comment:
Where is concurrent write happening on L127 ? It appears there are only reads on the array in question

Reply:
Nevermind, found the writes